app/routes/planet_routes.py

```diff
@@ -0,0 +1,113 @@
+from flask import Blueprint, abort, make_response, Response, request
+from app.models.planet import Planet
+from sqlalchemy import desc
+from ..db import db
+
+
+planets_bp = Blueprint("planets_bp", __name__, url_prefix="/planets")
+
+
+@planets_bp.get("")
+def get_all_planets():
+    query = db.select(Planet)
+
+    name_query = request.args.get("name")
+    if name_query:
+        query = query.where(Planet.name.ilike(f"%{name_query}%"))
+    
+    description_query = request.args.get("description")
+    if description_query:
+        query = query.where(Planet.description.ilike(f"%{description_query}%"))
+
+    diameter_query = request.args.get("diameter")
+    if diameter_query:
+        query = query.where(Planet.diameter.ilike(f"%{diameter_query}%"))
+
+    query = query.order_by(desc(Planet.diameter))
+    planets = db.session.scalars(query)
+
+    planet_response = []
+
+    for planet in planets:
+        planet_response.append(
+            {
+                "id": planet.id,
+                "name": planet.name,
+                "description": planet.description,
+                "diameter": planet.diameter
+            }
+        )
+    return planet_response
+
+
+@planets_bp.post("")
+def create_planet():
+    request_body = request.get_json()
+    name = request_body["name"]
+    description = request_body["description"]
+    diameter = request_body["diameter"]
+
+    new_planet = Planet(name=name, description=description, diameter=diameter)
+    db.session.add(new_planet)
+    db.session.commit()
+
+    response = {
+
+        "id": new_planet.id,
+        "name": new_planet.name,
+        "description": new_planet.description,
+        "diameter": new_planet.diameter
+    }
+    return response, 201
+
+
+@planets_bp.get("/<planet_id>")
+def get_one_planet(planet_id):
+    planet = validate_planet(planet_id) 
+
+    return {
+
+        "id": planet.id,
+        "name": planet.name,
+        "description": planet.description,
+        "diameter": planet.diameter
+    }
```

Comment:
Could call `to_dict` on planet here instead so creating this literal dictionary isn't happening three different times in your route file.